### DISCUSSION TITLE
랭킹 정보 조회가 안됩니다.

### DISCUSSION THREAD

**Discussion by Sonak0930** [naverz/zepeto-studio-kor#1043] [5. Q&A]
리더보드 생성및 랭킹 조회 튜토리얼을 따라서 성공했습니다.
그런데  어떤 데이터가 어떻게 저장되는지 모르겠습니다.

**Comment by ssm00** [comment]
안녕하세요 제가 이전에 리더보드를 생성했던 코드를 첨부합니다.
GetRanking함수 밖에서 다른 메소드로 랭킹데이터를 업데이트하는 것이 작동하지않아 GetRanking함수 안에 그대로 작성하였는데 그 부분을 참고하시면 좋을 것 같습니다.
저는 유니티 UI를 이용해 순위, 이름, 점수를 Text UI빈칸으로 30개를 미리 만들어두고  해당 클래스로 랭킹을 업데이트 해주는 방식으로 구현하였습니다.  코드는 UI에 랭킹을 업데이트하는 부분입니다.
코드에 궁금하신점이 있으시다면 물어봐주시고 도움이 되셨으면 좋겠네요
![1](https://user-images.githubusercontent.com/97657265/181066619-9a231ed3-2063-4b10-b75f-69937479a052.PNG)

**Reply by ssm00** [reply]
코드 위에 변수가 짤려서 재첨부해요
![1](https://user-images.githubusercontent.com/97657265/181067300-244b46ea-a991-4529-83ea-ecddae974851.PNG)
~

**Reply by Sonak0930** [reply]
감사합니다. lamda가 아니면 함수가 동작하지를 않더군요. 혹시 Score가 업데이트 되면 min이나 max값만 표현되는 걸까요?
SetScore로 랜덤한 값을 계속해서 넣어주고 있는데 가장 큰 값 / 가장 작은 값만 표시되는거 같아서요.

**Reply by ssm00** [reply]
네 처음 리더보드 만드실때 max,min 선택한 값으로 돼요 값이 하나만 나오는 건 아이디 당 1개의 값만 설정돼서 그러는것 같아요 저도 출시 전 테스트할때는 제 기록만 계속 최대값으로 갱신돼서 나오더라고요 월드 출시후에 확인해봤는데 정상적으로 랭킹생성되는것 같아요

**Reply by Sonak0930** [reply]
덕분에 잘 동작하고 있어요! 리더보드 공부하실때 하나하나 실험해보면서 알아가셨나요? prevRanking같은 argument는 어떻게 쓰는건지 잘 몰라서요. 실험해볼까 싶어요

**Reply by ssm00** [reply]
예시를 보고 사용해서 prevRanking은 잘 모르겠어요 하나하나 찍어보고 다른 질문들 참고하여 공부했었고 기본적으로 유니티에서 돌아가는 C#코드를 타입스크립트로 옮긴다 생각하고 만들었었던거같아요

**Reply by hyunjoon-park-metaspace** [reply]
혹시 에디터 내에서도 확인이 되시나요? 저도 리더보드를 구현중인데 어떠한 콜백도 호출되지 않고 있어서요. 감사합니다.

**Reply by Sonak0930** [reply]
네네 저는 만드는데 성공했습니다. 콜백이 호출되지 않는경우는 보통 parameter 세팅이 제대로 되지 않았거나 함수 호출이 되지 않았거나 여러가지 원인이 있을 수 있습니다. 먼저 리더보드가 생성이 되어있는지, 리더보드 id를 setScore와 getScore에 잘 넣고 있는지, 그리고 startRank는 1부터 시작하는지 확인해 보시면 좋을 거 같아요. 아 그리고 참고로 getResult는 lambda 식으로 작성하셔야 동작합니다. 그냥 메소드 정의하면 버그가 있어서 되지 않더라구요.

**Reply by hyunjoon-park-metaspace** [reply]
아마 메소드가 안 되시는건 `this.Method.bind(this)` 처럼 `this` 바인딩을 안 하셔서 그런 것으로 보입니다. startRank가 1부터 시작하는군요! 감사합니다 :)

**Comment by chlguswls32** [comment]
코드 공유해주셔서 감사합니다 ㅎㅎ

**Comment by jgw1413** [comment]
와... 정말 저 위에분 말대로 콜백 안에서는 프리팹 생성, 어떠한 함수 호출도 안돼서 저렇게 수정했더니 바로 됐습니다... 감사합니다
혹시나 저처럼 거의 2시간반이나 삽질하지 않도록 코드 공유합니다.
![캡처](https://user-images.githubusercontent.com/43944952/202395065-2947517a-e770-41dd-8b6c-0f3ad6f5ce69.PNG)
